File: sdk-python/examples/chat_streaming.py
"""Demonstrates streaming chat completions through C6 Guardrails.

Shows:
- Sending a chat completion request with streaming enabled.
- Iterating over chunks with async for and printing delta content.
- Using asyncio.run().
"""

import asyncio

from c6_guardrails import (
    ChatCompletionRequest,
    ChatMessage,
    GatewayClient,
    GatewayClientOptions,
)


async def main() -> None:
    client = GatewayClient(GatewayClientOptions(base_url="http://localhost:8082", api_key="ak_..."))

    request = ChatCompletionRequest(
        model="gpt-4o",
        messages=[ChatMessage(role="user", content="Tell me a short joke")],
    )

    async for chunk in client.chat_stream(request):
        for choice in chunk.choices:
            delta = choice.delta.get("content", "")
            if delta:
                print(delta, end="", flush=True)

    print()

    await client.close()


asyncio.run(main())

File: sdk-node/examples/chat-streaming.ts
/**
 * Chat streaming example — demonstrates chatStream() with a simple
 * prompt, for await...of iteration over chunks, printing delta
 * content, and counting total tokens from the final chunk usage.
 */

import { GatewayClient } from '@c6web/guardrails';

const client = new GatewayClient();

let totalTokens = 0;

for await (const chunk of client.chatStream({
  model: 'gpt-4o',
  messages: [{ role: 'user', content: 'Say hello in 5 words' }],
})) {
  for (const choice of chunk.choices) {
    if (choice.delta.content) {
      process.stdout.write(choice.delta.content);
    }
    if (choice.finish_reason) {
      console.log('\nFinish reason:', choice.finish_reason);
    }
  }
}
